src/lib/multicall/executeMulticall.ts

```diff
@@ -0,0 +1,274 @@
+  chainId: number,
+  request: TConfig,
+  priority: "urgent" | "background" = "urgent",
+  /**
+   * For debugging purposes, you can provide a name to the multicall request.
+   */
+  name?: string
+): Promise<MulticallResult<TConfig>> {
+  let groupNameMapping: {
+    // Contract address
+    [address: string]: {
+      // Unique call id based on contract address, method name and params
+      [callId: string]: {
+        // Human readable call group name
+        callGroupName: string;
+        // Human readable call name
+        callName: string;
+      }[];
+    };
+  } = {};
+
+  const { promise, resolve } = Promise.withResolvers();
+
+  const hook = (data: MulticallResult<any>) => {
+    const strippedRenamedData: MulticallResult<any> = {
+      success: data.success,
+      errors: {},
+      data: {},
+    };
+
+    for (const [contractAddress, contractResult] of entries(data.data)) {
+      if (!groupNameMapping[contractAddress]) {
+        continue;
+      }
+
+      for (const [callId, callResult] of entries(contractResult)) {
+        if (!groupNameMapping[contractAddress][callId]) {
+          continue;
+        }
+
+        const destinations = groupNameMapping[contractAddress][callId];
+
+        for (const { callGroupName, callName } of destinations) {
+          if (!strippedRenamedData.data[callGroupName]) {
+            strippedRenamedData.data[callGroupName] = {};
+          }
+
+          strippedRenamedData.data[callGroupName][callName] = callResult;
+        }
+      }
+    }
+
+    for (const [contractAddress, contractErrors] of entries(data.errors)) {
+      if (!groupNameMapping[contractAddress]) {
+        continue;
+      }
+
+      for (const [callId, error] of entries(contractErrors)) {
+        if (!groupNameMapping[contractAddress][callId]) {
+          continue;
+        }
+
+        const destinations = groupNameMapping[contractAddress][callId];
+
+        for (const { callGroupName, callName } of destinations) {
+          if (!strippedRenamedData.errors[callGroupName]) {
+            strippedRenamedData.errors[callGroupName] = {};
+          }
+
+          strippedRenamedData.errors[callGroupName][callName] = error;
+        }
+      }
+    }
+
+    resolve(strippedRenamedData);
+  };
+
+  for (const [callGroupName, callGroup] of entries(request)) {
+    if (!groupNameMapping[callGroup.contractAddress]) {
+      groupNameMapping[callGroup.contractAddress] = {};
+    }
+    for (const [callName, call] of entries(callGroup.calls)) {
+      if (!call) {
+        continue;
+      }
+
+      // To reduce duplicate calls, we hash the call data and use it as a unique identifier.
+      // There are two main reasons for this:
+      // 1. Single token backed pools have many pairs with the same method signatures
+      // 2. The majority of pools have USDC as the short token, which means they all have some common calls
+
+      const callId = stableHash([callGroup.contractAddress, call.methodName, call.params]);
```

Comment:
just in case, have you measured this call?

Reply:
no i did not.
but there is no need.
use swr uses it. In reality its not a hash. Just a concatenated and serialized view. concatenating 3-8 primitives is fast

Reply:
<img width="381" alt="image" src="https://github.com/user-attachments/assets/c7eef75d-1bef-4f56-bb90-e4bec523a4bf">
<img width="503" alt="image" src="https://github.com/user-attachments/assets/1884e832-4b86-4522-a9aa-d74065a2455a">

Reply:
https://github.com/vercel/swr/blob/1585a3e37d90ad0df8097b099db38f1afb43c95d/src/_internal/utils/serialize.ts

https://github.com/vercel/swr/blob/1585a3e37d90ad0df8097b099db38f1afb43c95d/src/core/use-swr.ts#L118